### DISCUSSION TITLE
Override at_object_creation on create_object?

### DISCUSSION THREAD

**Discussion by kovitikus** [evennia/evennia#2289] [Evennia Questions] [answered]
Let's say I add an attribute to my object in `at_object_creation`, perhaps `health = 100`. When I create the object using `evennia.utils.create.create_object`, if I pass in my kwarg `attributes = [(health, 500), ]`, which of the health attributes will actually take precedence?

I want to create a default value for an attribute, in case I'm not going to manually pass it in when I create something, but would like to have the option to override.

**Comment by Griatch** [comment, marked as answer]
The answer is that both will apply, but the `create_object`-supplied attributes will apply _last_ and potentially overwrite those in the hook. 

Think of it as the properties supplied at creation time (that is, in your `create.create_object` call) will take precedence over those that are 'hardcoded' in the `at_object_creation` hook - the idea is that you may want to tweak an object at-creation beyond the default set up in its typeclass.

This is also why prototypes/spawn will work nicely with any typeclass without you needing to remember if you left stuff in its `at_*_creation` hook to mess with the spawning.

**Reply by kovitikus** [reply]
Thanks, I understand. The `create_object` attributes kwarg is as if I'm executing `obj.db.health = 500` in code after the `at_object_creation` hook is run.
